guppylang/custom.py

```diff
@@ -53,7 +50,10 @@ def __init__(
     @property  # type: ignore[override]
     def ty(self) -> FunctionType:
         if self._ty is None:
-            return UnknownFunctionType()
+            # If we don't have a specified type, then the extension writer has to
+            # provide their own type-checking code. Therefore, it doesn't matter which
+            # type we return here since it will never be inspected.
+            return FunctionType([], NoneType())
```

Comment:
I got rid of `UnknownFunctionType` since it was very hacky. It's fine to return an arbitrary function type here since the actual checking logic is delegated to the extension writer